nuxeo-services/nuxeo-platform-directory/nuxeo-platform-directory-core/src/main/java/org/nuxeo/ecm/directory/DirectoryCSVLoader.java

```diff
@@ -107,13 +138,25 @@ public static void loadData(String dataFileName, char delimiter, Schema schema,
                     Object v = CSV_NULL_MARKER.equals(value) ? null : decode(field, value);
                     map.put(field.getName().getPrefixedName(), v);
                 }
-                loader.accept(map);
+                loaderWithLineno.accept(map, lineno);
             }
         } catch (IOException e) {
             throw new DirectoryException("Read error while reading data file: " + dataFileName, e);
         }
     }
 
+    protected static BiConsumer<Map<String, Object>, Integer> toLoaderEnrichedOnError(
+            Consumer<Map<String, Object>> loader) {
+        return (map, lineno) -> {
+            try {
+                loader.accept(map);
+            } catch (DirectoryException e) {
+                e.addInfo(String.format("Exception: %s happened on line: %s", e.getMessage(), lineno));
```

Comment:
I would just do `e.addInfo("At line: " + lineno)` as the exception message is already in the exception itself.